src/BuildingRegistry/Building/Building.cs

```diff
@@ -236,6 +236,30 @@ public void ChangeOutliningConstruction(ExtendedWkbGeometry extendedWkbGeometry)
                 buildingUnitsPosition));
         }
 
+        public void Demolish()
+        {
+            GuardRemovedBuilding();
+
+            if (BuildingGeometry.Method != BuildingGeometryMethod.MeasuredByGrb)
+            {
+                throw new BuildingHasInvalidGeometryMethodException();
+            }
+
+            if (BuildingStatus == BuildingStatus.Retired)
+            {
+                return;
+            }
+
+            GuardValidStatusses(BuildingStatus.Realized);
+
+            ApplyChange(new BuildingWasDemolished(BuildingPersistentLocalId));
+
+            foreach (var unit in _buildingUnits)
```

Comment:
I think the building unit events should come before the building event, see
https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry/Building/Building.cs#L99-L104
and
https://github.com/Informatievlaanderen/building-registry/blob/main/src/BuildingRegistry/Building/BuildingUnit.cs#L87-L95